app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/FractionInteractionViewModel.kt

```diff
@@ -64,23 +69,25 @@ class FractionInteractionViewModel private constructor(
 
   /** It checks the pending error for the current fraction input, and correspondingly updates the error string based on the specified error category. */
   override fun checkPendingAnswerError(category: AnswerErrorCategory): String? {
-    if (answerText.isNotEmpty()) {
-      when (category) {
-        AnswerErrorCategory.REAL_TIME -> {
+    when (category) {
+      AnswerErrorCategory.REAL_TIME -> {
+        if (answerText.isNotEmpty()) {
           pendingAnswerError =
             FractionParsingUiError.createFromParsingError(
               fractionParser.getRealTimeAnswerError(answerText.toString())
             ).getErrorMessageFromStringRes(resourceHandler)
+        } else {
+          pendingAnswerError = null
```

Comment:
My understanding is that the message should show on submit only, not while the user is typing, or waiting to type,  the answer. Otherwise we would show an error message in the resting state, when the user did not input any answer, in which case we could keep the submit button disabled.